File: src/main/java/com/beat/domain/performance/application/dto/performanceDetail/PerformanceDetailResponse.java
package com.beat.domain.performance.application.dto.performanceDetail;

import java.util.List;

public record PerformanceDetailResponse(
        Long performanceId,
        String performanceTitle,
        String performancePeriod,
        List<PerformanceDetailScheduleResponse> scheduleList,
        int ticketPrice,
        String genre,
        String posterImage,
        int runningTime,
        String performanceVenue,
        String performanceDescription,
        String performanceAttentionNote,
        String performanceContact,
        String performanceTeamName,
        List<PerformanceDetailCastResponse> castList,
        List<PerformanceDetailStaffResponse> staffList,
        int minDueDate,
        List<PerformanceDetailImageResponse> performanceImageList
) {
    public static PerformanceDetailResponse of(
            Long performanceId,
            String performanceTitle,
            String performancePeriod,
            List<PerformanceDetailScheduleResponse> scheduleList,
            int ticketPrice,
            String genre,
            String posterImage,
            int runningTime,
            String performanceVenue,
            String performanceDescription,
            String performanceAttentionNote,
            String performanceContact,
            String performanceTeamName,
            List<PerformanceDetailCastResponse> castList,
            List<PerformanceDetailStaffResponse> staffList,
            int minDueDate,
            List<PerformanceDetailImageResponse> performanceImageList
    ) {
        return new PerformanceDetailResponse(
                performanceId,
                performanceTitle,
                performancePeriod,
                scheduleList,
                ticketPrice,
                genre,
                posterImage,
                runningTime,
                performanceVenue,
                performanceDescription,
                performanceAttentionNote,
                performanceContact,
                performanceTeamName,
                castList,
                staffList,
                minDueDate,
                performanceImageList
        );
    }
}

File: apis/src/main/kotlin/com/beat/apis/performance/api/response/PerformanceDetailResponse.kt
package com.beat.apis.performance.api.response

import com.beat.global.support.jackson.CdnImageUrl

data class PerformanceDetailResponse(
    val performanceId: Long?,
    val performanceTitle: String?,
    val performancePeriod: String?,
    val scheduleList: List<PerformanceDetailScheduleResponse>,
    val ticketPrice: Int,
    val genre: String?,
    @field:CdnImageUrl val posterImage: String?,
    val runningTime: Int,
    val performanceVenue: String?,
    val roadAddressName: String?,
    val placeDetailAddress: String?,
    val latitude: String?,
    val longitude: String?,
    val performanceDescription: String?,
    val performanceAttentionNote: String?,
    val performanceContact: String?,
    val performanceTeamName: String?,
    val castList: List<PerformanceDetailCastResponse>,
    val staffList: List<PerformanceDetailStaffResponse>,
    val minDueDate: Int,
    val performanceImageList: List<PerformanceDetailImageResponse>,
)
